src/components/CommentImageButton.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import Icon from './Icon'
+import { notify } from '../actions'
+import { uploadImage } from '../actions/uploadImage'
+import { imageUploadSettings } from '../models/comment'
+import { createComment } from '../actions/comments'
+
+const CommentImageButton = ({ postId }, { dispatch, currentUser }) => {
+  const sendImage = () =>
+    dispatch(uploadImage(imageUploadSettings(currentUser.id, postId)))
+    .then(({ payload, error }) => {
+      if (error) return dispatch(notify('There was a problem sending your image. Please try again in a moment', {type: 'error'}))
+      return dispatch(createComment({postId, imageUrl: payload}))
+    })
+
+  return <a className='send-image' onClick={sendImage}><Icon name='Camera' /></a>
+}
+CommentImageButton.contextTypes = {
+  dispatch: React.PropTypes.func, currentUser: React.PropTypes.object
```

Comment:
Do this above:
```javascript
import React, { PropTypes } from 'react'
```
and then here: 
```javascript
 dispatch: PropTypes.func, ...
```
I prefer that both to what you've done here and to fully resolving down to just "func".  Again, just my own opinions.